lib/features/details/presentation/view/movie_details_page.dart

```diff
@@ -89,31 +129,46 @@ class _MovieDetailsPageState extends State<MovieDetailsPage> {
                 const SizedBox(height: 12),
                 Align(
                   alignment: Alignment.centerLeft,
-                  child: Container(
-                    height: 45,
-                    width: 120,
-                    decoration: BoxDecoration(
-                        border: Border.all(color: Colors.white),
-                        borderRadius: BorderRadius.circular(12)),
-                    child: const Padding(
-                      padding: EdgeInsets.all(8.0),
-                      child: Row(
-                        children: [
-                          Icon(
-                            Icons.check,
-                            color: Colors.white,
-                          ),
-                          SizedBox(
-                            width: 5,
+                  child: ValueListenableBuilder(
+                      valueListenable: movieInWatchList,
+                      builder: (context, value, _) {
+                        return GestureDetector(
+                          onTap: () {
+                            saveMovie(widget.movie);
+                          },
+                          child: Container(
+                            height: 45,
+                            width: 120,
+                            decoration: BoxDecoration(
+                                border: Border.all(color: Colors.white),
+                                borderRadius: BorderRadius.circular(12)),
+                            child: Padding(
+                              padding: const EdgeInsets.all(8.0),
+                              child: Row(
+                                children: [
+                                  value
```

Comment:
You can condense this to:

```dart
Icon(
   value ? Icons.check : Icons.add,
   color: Colors.white,
 )
```

Leverage the commonness!

With your `ValueListenableBuilder` in the best place, it'll be:

```dart
ValueListenableBuilder(
   valueListenable: movieInWatchList,
   builder: (context, value, _) {
        return Icon(
           value ? Icons.check : Icons.add,
           color: Colors.white,
         );
   }
)
```

This way, you're only ever rebuilding the icon when `movieInWatchList` changes and nothing else